### ISSUE TITLE
Proper way to use VCPKG with Intel C++ Compiler 19.0 (ICL)

### ISSUE THREAD

**Issue by YenForYang** [microsoft/vcpkg#4487]
Currently I'm using this hack --- modifying `scripts/toolchains/windows.cmake` by adding

```cmake
set(CMAKE_GENERATOR_TOOLSET "Intel C++ Compiler 19.0" CACHE STRING "Platform Toolset" FORCE)
```

And manually modifying the CMAKE_C_FLAGS and CMAKE_CXX_FLAGS in the same file each time I build something...

I also have to disable PREFER_NINJA in the package's portfile.

I attempted adding the line to a triplet file instead, but I believe it didn't work (99% sure).

 Is there a way to integrate ICL with Vcpkg?

**Comment by Voskrese** [comment]
xmm...
example
Clang/C2 can be specified as the Toolset -https://github.com/Microsoft/CMake/releases

`cmake -G "Visual Studio 14 2015" -T v140_clang_3_7 ...`

Or for a specific project, the toolset can be overridden using the `VS_PLATFORM_TOOLSET_OVERRIDE` variable as in:

`set_property(TARGET TestClang PROPERTY VS_PLATFORM_TOOLSET_OVERRIDE "v140_clang_3_7")`

**Comment by YenForYang** [comment]
What's the proper way to set the Intel C++ Compiler as the generator toolset globally (for vcpkg)?

I've actually been using a different approach (that very likely will lead to problems in the future)--involving monkey patching files in the `scripts` folder (i.e. I don't want to end up changing even single portfile).
I'm almost certain I'm doing it wrong (basically playing CMake script kiddy--have no idea what I'm doing)

So far, I've changed `scripts/vcpkg_configure_cmake.cmake` and `scripts/toolchains/windows.cmake` :

1. I've kept the below line in `toolchains/windows.cmake` (mainly because I have no idea where it ought to go at this point)
```cmake
set(CMAKE_GENERATOR_TOOLSET "Intel C++ Compiler 19.0" CACHE STRING "Platform Toolset" FORCE)
```
2. In `scripts/vcpkg_configure_cmake.cmake`, I've changed the following since the Ninja Generator doesn't configure with the Intel C++ Compiler 19.0 as the generator toolset:
```cmake
set(NINJA_CAN_BE_USED OFF)  # was ON 
```
3.  Finally (in the same file), I did:
```cmake
    set(rel_command
        ${CMAKE_COMMAND} ${_csc_SOURCE_PATH} "${_csc_OPTIONS}" "${_csc_OPTIONS_RELEASE}"
        -G ${GENERATOR}
        -T "Intel C++ Compiler 19.0"  # NOTE THIS LINE
        -DCMAKE_BUILD_TYPE=Release
        -DCMAKE_INSTALL_PREFIX=${CURRENT_PACKAGES_DIR})
    set(dbg_command
        ${CMAKE_COMMAND} ${_csc_SOURCE_PATH} "${_csc_OPTIONS}" "${_csc_OPTIONS_DEBUG}"
        -G ${GENERATOR}
        -T "Intel C++ Compiler 19.0"  # NOTE THIS LINE
        -DCMAKE_BUILD_TYPE=Debug
        -DCMAKE_INSTALL_PREFIX=${CURRENT_PACKAGES_DIR}/debug)
```

I have a feeling that I could use `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` instead of modifying `windows.cmake` (but I have no idea how to implement that).

I'll also mention that I'm using Visual Studio 2019 on my machine (Intel Parallel Studio 2019 Update 4, was released 5-20-2019 which added support for Visual Studio 2019). Versions before Version 19.0.4.228 don't have official support (latest version is Version 19.0.4.245).
